### ISSUE TITLE
Development Workflow for SpaceROS

### ISSUE THREAD

**Issue by franklinselva** [space-ros/space-ros#211]
### Feature description

Since the `dev` variant of the Space ROS image is prepared at the moment, I wanted to see what workflows should be integrated into the development process. I can see a few issues related to this which I will list below. I want to see if there are another points to be added related to the topic.

 - [ ] #198 
 - [ ] #183 
 - [ ] #80 & to image?
 - [ ] #30 
 - [ ] #54 
 - [ ] Static analysers and Formatting tools for CPP development (cppcheck, clangtidy, clangformat)
 - [ ] #7 
 - [ ] Docker lint and formatting setup
 - [ ] Python formatting, linting and static analysis tools
 - [ ] Add devcontainer configuration for space-ros images (from [here](https://github.com/space-ros/space-ros/pull/186#issuecomment-2300140414))

### Implementation considerations

The implementation should serve as a common ground for the CI to validate the standards

Is there any repository that can be referenced for the above list to fetch the current status?

**Comment by asimonov** [comment]
good idea. i would like to understand what happens now with the static analysers and then with the results of their runs (apart from saving for later reference)
